compiler/src/dotty/tools/dotc/transform/SyntheticMethods.scala

```diff
@@ -59,7 +59,7 @@ class SyntheticMethods(thisPhase: DenotTransformer) {
   def syntheticMethods(clazz: ClassSymbol)(implicit ctx: Context): List[Tree] = {
     val clazzType = clazz.appliedRef
     lazy val accessors =
-      if (isDerivedValueClass(clazz)) clazz.paramAccessors
+      if (isDerivedValueClass(clazz)) clazz.paramAccessors.take(1) // Tail parameters can only be `unused`
```

Comment:
??? Did not understand. Can you give an example?

Reply:
Value classes must have exactly one parameter, but we can add additional ones if they are `unused`. We already check that the first parameter list is not `unused` (same for case classes).
```scala
class Foo(v: Int)(unused ev: Ev) extends AnyVal
```
When we have that, we need to ignore the `unused` parameters when generating synthetic methods. In the case of a value class we know that the only the first parameter is needed.